diff --git a/node_modules/@semantic-release/npm/lib/definitions/constants.js b/node_modules/@semantic-release/npm/lib/definitions/constants.js
index 6c651a6..ad9d5d4 100644
--- a/node_modules/@semantic-release/npm/lib/definitions/constants.js
+++ b/node_modules/@semantic-release/npm/lib/definitions/constants.js
@@ -1,4 +1,5 @@
 export const OFFICIAL_REGISTRY = "https://registry.npmjs.org/";
 
+export const CIRCLECI_PROVIDER_NAME = "CircleCI";
 export const GITHUB_ACTIONS_PROVIDER_NAME = "GitHub Actions";
 export const GITLAB_PIPELINES_PROVIDER_NAME = "GitLab CI/CD";
diff --git a/node_modules/@semantic-release/npm/lib/trusted-publishing/token-exchange.js b/node_modules/@semantic-release/npm/lib/trusted-publishing/token-exchange.js
index 899560b..ed57c09 100644
--- a/node_modules/@semantic-release/npm/lib/trusted-publishing/token-exchange.js
+++ b/node_modules/@semantic-release/npm/lib/trusted-publishing/token-exchange.js
@@ -1,8 +1,13 @@
 import { getIDToken } from "@actions/core";
 import envCi from "env-ci";
+import { promisify } from "util";
+import { exec } from "child_process";
+
+const execAsync = promisify(exec);
 
 import {
   OFFICIAL_REGISTRY,
+  CIRCLECI_PROVIDER_NAME,
   GITHUB_ACTIONS_PROVIDER_NAME,
   GITLAB_PIPELINES_PROVIDER_NAME,
 } from "../definitions/constants.js";
@@ -28,6 +33,18 @@ async function exchangeIdToken(idToken, packageName, logger) {
   return undefined;
 }
 
+async function exchangeCircleCIToken(packageName, logger) {
+  const idToken = process.env.NPM_ID_TOKEN;
+
+  logger.log("Verifying OIDC context for publishing from CircleCI");
+
+  if (!idToken) {
+    return undefined;
+  }
+
+  return exchangeIdToken(idToken, packageName, logger);
+}
+
 async function exchangeGithubActionsToken(packageName, logger) {
   let idToken;
 
@@ -60,6 +77,10 @@ async function exchangeGitlabPipelinesToken(packageName, logger) {
 export default function exchangeToken(pkg, { logger }) {
   const { name: ciProviderName } = envCi();
 
+  if (CIRCLECI_PROVIDER_NAME === ciProviderName) {
+    return exchangeCircleCIToken(pkg.name, logger);
+  }
+
   if (GITHUB_ACTIONS_PROVIDER_NAME === ciProviderName) {
     return exchangeGithubActionsToken(pkg.name, logger);
   }
